Makefile

```
@@ -17,13 +17,25 @@ ifeq ($(origin COMPOSER_AUTH_PASSWORD), undefined)
   $(error COMPOSER_AUTH_PASSWORD is undefined.)
 endif
 
-docker_compose ?= docker compose
+search_compose_file = $(shell \
+	bash -c 'v1="$$(echo "$(MAGENTO_VERSION)" | sed "s/-p[0-9]\+//")"; \
+	v2="2.4.8"; \
+	if [ "$$(printf "%s\n%s" "$$v1" "$$v2" | sort -V | head -n1)" = "$$v2" ]; then \
+		echo docker-compose.opensearch.yml; \
+	else \
+		echo docker-compose.elasticsearch.yml; \
+	fi')
```

I prefer this form, but up to you, as yours works:

```suggestion
search_compose_file = $(shell \
	bash -c 'MAGENTO_VERSION=$(MAGENTO_VERSION); \
		echo -e "2.4.8\n$${MAGENTO_VERSION%-*}" | sort -cV &> /dev/null && \
		echo docker-compose.opensearch.yml || \
		echo docker-compose.elasticsearch.yml')
```